### PR TITLE
Homogenize PF metrics naming

### PR DESCRIPTION
**Pull request by yue9944882**
/sig api-machinery
/kind cleanup

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by lavalamp** [comment]
/lgtm
/approve

cc @logicalhan

**Review by logicalhan — COMMENTED** [review]
I don't think these are going to work. These used the unwrapped prometheus metric descriptors. We need to use the metric descriptors from component-base instead.

**Review by logicalhan — COMMENTED** [review]
We should move the metric declarations into component base.

File: staging/src/k8s.io/apiserver/pkg/util/flowcontrol/metrics/metrics.go [hunk 1]
```diff
@@ -52,7 +52,7 @@ var (
 		prometheus.CounterOpts{
 			Namespace: namespace,
 			Subsystem: subsystem,
-			Name:      "rejectedRequests",
+			Name:      "rejected_requests",
```
Comment (brancz) [inline staging/src/k8s.io/apiserver/pkg/util/flowcontrol/metrics/metrics.go]: This should according to our instrumentation guideline also have the _total suffix.

**Comment by RainbowMango** [comment]
Since this has been merged, I [created an issue](https://github.com/kubernetes/kubernetes/issues/88426) to track this. 

And I found these metrics were introduced 4 months ago, maybe after our first round migration.
We should avoid things like this, and this reminds me of the tasks we haven't done.
We still have some metrics haven't migrated to component-base, and after that, we can restrict direct Prometheus reference. 

I'll start second round migration soon.

**Comment by yue9944882** [comment]
thanks for helping, plz let me if i misses anything in the follow-up: https://github.com/kubernetes/kubernetes/pull/88609

**Pull request merged**